SU2_CFD/src/iteration_structure.cpp

```diff
@@ -51,34 +51,34 @@ CIteration::CIteration(CConfig *config) {
 
 CIteration::~CIteration(void) { }
 
-void CIteration::SetGrid_Movement(CGeometry ****geometry_container,
-          CSurfaceMovement **surface_movement,
-          CVolumetricMovement ***grid_movement,
-          CFreeFormDefBox ***FFDBox,
-          CSolver *****solver_container,
-          CConfig **config_container,
-          unsigned short val_iZone,
-          unsigned short val_iInst,
+void CIteration::SetGrid_Movement(CGeometry **geometry,
+          CSurfaceMovement *surface_movement,
+          CVolumetricMovement *grid_movement,
+          CSolver ***solver,
+          CConfig *config,
           unsigned long IntIter,
           unsigned long ExtIter)   {
 
   unsigned short iDim;
-  unsigned short Kind_Grid_Movement = config_container[val_iZone]->GetKind_GridMovement();
+  unsigned short Kind_Grid_Movement = config->GetKind_GridMovement();
   unsigned long nIterMesh;
   unsigned long iPoint;
   bool stat_mesh = true;
-  bool adjoint = config_container[val_iZone]->GetContinuous_Adjoint();
-  bool harmonic_balance = (config_container[val_iZone]->GetUnsteady_Simulation() == HARMONIC_BALANCE);
-  bool discrete_adjoint = config_container[val_iZone]->GetDiscrete_Adjoint();
+  bool adjoint = config->GetContinuous_Adjoint();
+  bool harmonic_balance = (config->GetUnsteady_Simulation() == HARMONIC_BALANCE);
+  bool discrete_adjoint = config->GetDiscrete_Adjoint();
 
   /*--- Only write to screen if this option is enabled ---*/
-  bool Screen_Output = config_container[val_iZone]->GetDeform_Output();
+  bool Screen_Output = config->GetDeform_Output();
+  
+  unsigned short val_iZone = config->GetiZone();
+  unsigned short val_iInst = config->GetiInst();
```

Comment:
Can you really please not do this? It is not enough to call a method one needs to remember to set things in config first?
And in this case it is not even the immediate caller of the function that sets the values... Please convince me otherwise but honestly this seems like a terrible idea.
(I known there is precedent with the Runtime stuff...)

Reply:
I don't like it either ... But now that I looked at it again, we can just pass the iInst using the ExtIter parameter (its copied as ExtIter anyway)

Reply:
I removed it in that routine. This way also the "if (harmonic_balance)" was not necessary anymore.

Reply:
Thanks